### DISCUSSION TITLE
Single operator returns undefined

### DISCUSSION THREAD

**Discussion by DusanDjordjic** [ReactiveX/rxjs#6798] [Q&A] [answered]
Greetings,
I am was going through RxJS operators documentation and I noticed that it says that the single operator will throw a `NotFoundError`  if no value matches the predicate but it sends `undefined` to the next function. 

Here is the code that I wrote: 

``` ts
  const source = of(
    { name: 'Dusan' },
    { name: 'Jovana' },
    { name: 'Jovan' },
    { name: 'Aleksandar' }
  );

source.pipe(single((person) => person.name.startsWith('N'))).subscribe({
    next: (value) => {
      console.log({value});
    },
    error: (err) => {
      console.log({err});
    },
});

```

My question is, is it a bug or just an error in the documentation?

**Comment by josepot** [comment, marked as answer]
What version of rxjs are you using? :thinking: 

I [get the `NotFoundError` on version 7.5.2](https://codesandbox.io/s/sweet-pascal-xbzx6?file=/src/index.js). Could you please share a codesandbox/stackblitz that reproduces the issue?

**Comment by DusanDjordjic** [comment]
My bad. I loaded an old project with rxjs 6.6 and I didn't check the version,  so it didn`t work. 

Sorry for taking your time.  😅

**Reply by josepot** [reply]
> Sorry for taking your time. sweat_smile

Nothing to be sorry about.

In fact, by the looks of it: this **is** [a bug on version 6](https://codesandbox.io/s/optimistic-babbage-07r6s?file=/src/index.js), because [the docs of v6](https://v6.rxjs.dev/api/operators/single) have the same statement. I'm not sure whether v6 is still under maintenance, but if it is, then I think that this issue should be addressed.
